src/Spanner/Database.php

```diff
@@ -643,10 +652,15 @@ public function transaction(array $options = [])
      *           `false`.
      * }
      * @return mixed The return value of `$operation`.
-     * @throws \RuntimeException
+     * @throws \RuntimeException If a transaction is not committed or rolled back.
+     * @throws \BadMethodCallException If a nested transaction is created.
      */
     public function runTransaction(callable $operation, array $options = [])
     {
+        if ($this->isRunningTransaction) {
```

Comment:
can we prevent read only transactions also from being nested inside a read write transaction?